osh/cmd_eval.py

```diff
@@ -1724,18 +1740,20 @@ def _DoTimeBlock(self, node):
 
         return status
 
-    def _DoRedirect(self, node, cmd_st):
-        # type: (command.Redirect, CommandStatus) -> int
+    def _DoRedirects(self, redirects):
```

Comment:
I think this function could use a different name ... now it is a helper function for

- `_DoRedirect()` - which is part of the big `command_t` switch, for the `command.Redirect`
- `_DoSimple()` - the `command.Simple` case

There is _EvalRedirect

I guess this is NOT just _EvalRedirects

Maybe we can call it `_EvalAndApply(redirects)` or something

I think the `_Do*` names are better for the big _Dispatch() switch

Reply:
Oh now I see it could be `_EvalAndPushRedirects`  - because it only does the "first half", not the undoing of the redirects



(what annoys me a bit is that the API is not a symmetric Push/Pop because of error handling, but that is separate from this change!)